All_Relationships/5.sentence-prediction-analysis.py

```diff
@@ -168,30 +173,22 @@
 plt.legend(loc="lower right")
 
 
-# In[13]:
-
-
-for model in ["RNN_1_Predictions", "RNN_10_Predictions"]:
-    matrix = confusion_matrix(model_marginals["True_Labels"], model_marginals[model])
-    chart1, ax1 = plt.subplots()
-    plt.title(model)
-    ax = sns.heatmap(matrix, annot=True, fmt="d", cmap="BuGn", xticklabels=["No","Yes"], yticklabels=["No", "Yes"], ax=ax1)
-    ax.set_xlabel("Predicted Labels")
-    ax.set_ylabel("True Labels")
-
-
 # ## LSTM Model Analysis
 
 # In[14]:
 
-
-condition = (model_marginals["RNN_10_Predictions"] == 1)&(model_marginals["True_Labels"] == -1)
-indicies = set(model_marginals[condition].sort_values("RNN_10_Marginals", ascending=False).index)
-model_marginals[condition].sort_values("RNN_10_Marginals", ascending=False).head(10)
+marginal_criteria = "RNN_100_Marginals"
+model_predictions = model_marginals[marginal_criteria].apply(lambda x: 1 if x > 0.5 else -1)
+model_predictions.head(10)
 
 
 # In[15]:
 
+condition = (model_predictions == 1)&(model_marginals["True_Labels"] == -1)
+model_marginals[condition].sort_values(marginal_criteria, ascending=False).head(10)
```

Comment:
The following is shorter, if supported by your env:

```py
model_marginals.query("model_predictions == 1 and True_Labels == -1")
```

Reply:
Gotcha!